src/System.Runtime.InteropServices/tests/System/Runtime/InteropServices/Marshal/StringMarshalingTests.cs

```diff
@@ -0,0 +1,198 @@
+                                    "This is a very long string as it repeats itself. " +
+                                    "This is a very long string as it repeats itself. " +
+                                    "This is a very long string as it repeats itself. " +
+                                    "This is a very long string as it repeats itself. " +
+                                    "This is a very long string as it repeats itself. " +
+                                    "This is a very long string as it repeats itself.",
+                                    "This \n is \n a \n multiline \n string",
+                                    "This \0 is \0 a \0 string \0 with \0 nulls",
+                                    "\0string",
+                                    "string\0",
+                                    "\0\0\0\0\0\0\0\0"
+                                    };
+
+        [Fact]
+        public void StringToBStrToString()
+        {
+            foreach (String ts in TestStrings)
+            {
+                IntPtr BStr = Marshal.StringToBSTR(ts);
+                String str = Marshal.PtrToStringBSTR(BStr);
+                Assert.Equal(ts, str);
+                Marshal.FreeBSTR(BStr);
+            }
+        }
+
+        [Fact]
+        public unsafe void SecureStringToBSTRToString()
+        {
+            foreach (String ts in TestStrings)
+            {
+                SecureString secureString = new SecureString();
+                foreach (char character in ts)
+                {
+                    secureString.AppendChar(character);
+                }
+
+                IntPtr BStr = IntPtr.Zero;
+                String str;
+
+                try
+                {
+                    BStr = Marshal.SecureStringToBSTR(secureString);
+                    str = Marshal.PtrToStringBSTR(BStr);
+                }
+                finally
+                {
+                    if (BStr != IntPtr.Zero)
+                    {
+                        Marshal.ZeroFreeBSTR(BStr);
+                    }
+                }
+                Assert.Equal(ts, str);
+            }
+        }
+
+        [Fact]
+        public void StringToCoTaskMemAnsiToString()
+        {
+            foreach (String ts in TestStrings)
+            {
+                if (ts.Contains("\0"))
+                    continue; //Skip the string with nulls case
+
+
+                IntPtr AnsiStr = Marshal.StringToCoTaskMemAnsi(ts);
+                String str = Marshal.PtrToStringAnsi(AnsiStr);
+
+                Assert.Equal(ts, str);
+                if (ts.Length > 0)
+                {
+                    String str2 = Marshal.PtrToStringAnsi(AnsiStr, ts.Length - 1);
+                    Assert.Equal(ts.Substring(0, ts.Length - 1), str2);
+                }
+                Marshal.FreeCoTaskMem(AnsiStr);
+            }
+        }
+
+        [Fact]
+        public void StringToCoTaskMemUniToString()
```

Comment:
Dup of https://github.com/dotnet/corefx/blob/7ae1a252d7e68c5513d2658de7a401c37e9b0504/src/System.Runtime.InteropServices/tests/System/Runtime/InteropServices/MarshalTests.cs#L72 ?

Reply:
I think not, Marshal.StringToCoTaskMemUni & Marshal.SecureStringToCoTaskMemUnicode

Reply:
Misread it.